library/src/main/java/com/getstream/sdk/chat/rest/response/ChannelState.java

```diff
@@ -355,6 +355,10 @@ public Message getLastMessage() {
     }
 
     public void setLastMessage(Message lastMessage) {
+        if (lastMessage.getDeletedAt() != null){
```

Comment:
this kind of caching (`lastMessage`) has no reason to exist:

- performance benefit is close to 0 (getting the last entry in a ordered list, takes a couple loops)
- complexity + chances of getting bugs gets higher

I suggest we merge `computeLastMessage` with `getLastMessage`; drop the cached `lastMessage` and `setLastMessage`. Performance wise we will never notice this

Reply:
I have tried to do this with removing `lastMessage`.
(Removed `lastMessage` and `getter` & `setter` functions for that.)
But `ChannelListDiffCallBack` couldn't identify updates(or deletes) of `lastMessage` by `computeLastMessage` since `messages` of ChannelState had been updated already and `computeLastMessage` returns same results for `old` and `new` `ChannelState`. So I think we need  buffer param: `lastMessage` in `ChannelState`